src/internal/macros.rs

```diff
@@ -157,10 +157,11 @@ macro_rules! bitflags {
             )*
         }
     ) => {
+        $(#[$outer])*
+        $vis struct $BitFlags($T);
+
         bitflags::bitflags! {
-            #[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
-            $(#[$outer])*
-            $vis struct $BitFlags: $T {
+            impl $BitFlags: $T {
```

Comment:
How come this was changed to declare the struct outside the call to the macro? This also goes for `Permissions`, which doesn't use the internal macro because of conflicting serde impls.

Reply:
After running `cargo expand`, I see the difference is that if declaring the struct inline, you get:
```rust
pub struct MessageFlags(<MessageFlags as ::bitflags::__private::PublicFlags>::Internal)
```
whereas doing it like this just gets you `pub struct MessageFlags(u64)` as normal. However, maybe instead you could add manual impls of `TypeSize` to the structs instead? It would mean that `Permissions` also gets a separate manual impl. Or as an alternative, get the `typesize` crate to work with associated types, or whatever the actual limitation here is. The `PublicFlags` trait I think is publically exported (despite the `__private` name there), so I don't think visibility is the problem.

Reply:
That would not be possible. I cannot blanket implementation
```rs
impl<F> TypeSize for F
where
    F: bitflags::Flags,
    F::Bits: TypeSize,
{
}
```
as this leads to orphan rule stuff where bitflags could technically add an implementation of Flags for `Option` or `Vec` and break typesize.

I cannot "fix" the derive in any way either to handle the internal type as that would require deriving TypeSize on the private bitflags internal type.

I could add a custom implementation in serenity, but that seems unnecessary when it currently works fine.